crates/builder/src/graph/adj_list.rs

```diff
@@ -59,12 +92,119 @@ impl<NI: Idx> AdjacencyList<NI, ()> {
             std::mem::align_of::<Target<NI, ()>>(),
             std::mem::align_of::<NI>()
         );
-
-        let targets = self.edges[node.index()].as_slice();
-
         // SAFETY: The types Target<T, ()> and T are verified to have the same
         //         size and alignment.
-        unsafe { std::slice::from_raw_parts(targets.as_ptr().cast(), targets.len()) }
+        unsafe { std::slice::from_raw_parts(self.targets.as_ptr().cast(), self.targets.len()) }
+    }
+}
+
+pub struct TargetsIter<'slice, NI: Idx> {
+    _targets: Targets<'slice, NI>,
+    slice: std::slice::Iter<'slice, NI>,
+}
+
+impl<'slice, NI: Idx> TargetsIter<'slice, NI> {
+    pub fn as_slice(&self) -> &'slice [NI] {
+        self.slice.as_slice()
+    }
+}
+
+impl<'slice, NI: Idx> IntoIterator for Targets<'slice, NI> {
+    type Item = &'slice NI;
+
+    type IntoIter = TargetsIter<'slice, NI>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        let slice = self.as_slice();
+        TargetsIter {
+            _targets: self,
+            slice: slice.iter(),
+        }
+    }
+}
+
+impl<'slice, NI: Idx> Iterator for TargetsIter<'slice, NI> {
+    type Item = &'slice NI;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.slice.next()
+    }
+}
+
+impl<NI: Idx> AdjacencyList<NI, ()> {
+    #[inline]
+    pub(crate) fn targets<'list, 'slice>(&'list self, node: NI) -> Targets<'slice, NI>
+    where
+        'list: 'slice,
```

Comment:
What was the reason again for the two lifetimes?
From what I understand, [`lock`](http://doc.rust-lang.org/1.73.0/std/sync/struct.Mutex.html#method.lock) is bound to the lifetime of `self`, which in this case here is `'list`, so using `Targets<'list, NI>` could also work, maybe? Did you run into any troubles down the road with this?

Reply:
You are right, I don't need the additional lifetime + constraint. During dev, it was necessary, but I don't recall why exactly. I changed it to 

```rust
pub(crate) fn targets(&self, node: NI) -> Targets<'_, NI> { ... }
```